### ISSUE TITLE
Partially electronic toll booths

### ISSUE THREAD

**Issue by SubaruSVX** [CitiesSkylinesMods/TMPE#803]
### Describe your idea
In some countries (China in particular) have electronic tollbooths and have staff on hand to use them - so I'm wondering... could we do the same with TMPE?

### Notes or questions?
Couple of problems:
1. No one as far as I'm aware has done so (so we'll have to somehow get that to work)
2. Normally, toll booths are one or the other - so this is a rare sighting unless another country does this sort of thing.

**Comment by aubergine10** [comment]
You can use **Automated Toll Booths** city/district policy https://skylines.paradoxwikis.com/Policies

**Comment by aubergine10** [comment]
See also: https://github.com/CitiesSkylinesMods/TMPE/wiki/City-and-District-Policies

**Comment by SubaruSVX** [comment]
Yeah, sorry, @aubergine10 - my head has been in a bit of a slump XD

**Comment by aubergine10** [comment]
Ah, now I get it! Yeah, nice idea! Now that we've got stuff like lane highlighting it might be possible - we could toggle 'express lane' on lane-by-lane basis, effectively setting 'automated toll' policy for that specific lane. Non-express lanes would still force vehicles to pause and pay.

I don't know what the code is like or what performance impacts so it'll have to wait until we get through the current round of updates.

**Comment by aubergine10** [comment]
Noticed something the other day that `NetLane` structs have a ticket price. I've not had chance to do much investigating, but it would appear that each lane can have it's own ticket price.

![image](https://user-images.githubusercontent.com/1386719/79826509-d7ab9980-8393-11ea-915a-511b30d76b30.png)

![image](https://user-images.githubusercontent.com/1386719/79826551-fb6edf80-8393-11ea-876c-34d20f9bbca0.png)

With TMPE we already have ability to select lanes (lane highlighter) so maybe it could work something like this:

1. Select building in no-tool mode (where TMPE toolbar is visible, but no tool selected)
2. Click a toll booth building (or road?)
3. UI/overlays appear to allow customisation

The visualisations could be an extension (extra overlay icons) to vehicle restricitons overlays:

* User can now dedicate certain lanes to certain vehicle types
* Cost icon toggles between 0, $, $$, $$$ (mod option to set currency?) to quickly set cost for that lane
* Mode icon toggles between wifi icon (no stopping), cash icon (stop and pay) and some other icon depicting 'auto' (based on city/district policy).

For example, you could create a cargo lane which is $$$, a city service lane which is free and cashless, etc.

The only thing then is how to handle cashless lanes not stopping? Pathfinder and/or vehicle AI must handle that already, as the automated toll district policy has to take effect somwhere. So we just need to inject a lane-specific flag in to that somehow.

There are some flags for lanes, and `Merge` might be potential candidate depending on how it's used:

![image](https://user-images.githubusercontent.com/1386719/79828589-e779ac80-8398-11ea-83f2-8432d3c50d0e.png)

Otherwise, we could potentially use the `Yield` flags maybe? Whatever, if we can use an existing `NetLane` flag then the game will automatically handle persistence for us and it will almost certainly be readily available to the pathfinder and vehicle AIs which will minimise performance impacts.

**Comment by aubergine10** [comment]
So, I checked all flags on netlane, they are basically all used even by the lanes of a small city:

`1111110111111101`

That's cumlative `OR` of all flags; the two empty flags were `Deleted` (as expected) and `Stop2` (I didn't add any tram stops yet).

So it seems a flags extension will be required, which in turn will also facilitate other things such as #515. It will take some time to try and create a coherent data model.
